ui/pages/defi/components/defi-details-page.tsx

```diff
@@ -0,0 +1,139 @@
+import DefiDetailsList, {
+  PositionTypeKeys,
+  ProtocolTokenWithMarketValue,
+} from './defi-details-list';
+
+const useExtractUnderlyingTokens = (
+  positions?: ProtocolTokenWithMarketValue[][],
+) =>
+  useMemo(() => {
+    if (!positions) {
+      return [[]];
+    }
+
+    return positions.map((group) =>
+      group.flatMap((position) => position.tokens.map((token) => token)),
+    );
+  }, [positions]);
+
+const DeFiPage = () => {
+  const { chainId, protocolId } = useParams<{
+    chainId: '0x' & string;
+    protocolId: string;
+  }>() as { chainId: '0x' & string; protocolId: string };
+
+  const defiPositions = useSelector(getDefiPositions);
+  const selectedAccount = useSelector(getSelectedAccount);
+
+  const history = useHistory();
+  const t = useI18nContext();
+  const { privacyMode } = useSelector(getPreferences);
+
+  const protocolPosition =
+    defiPositions[selectedAccount.address]?.[chainId]?.protocols[protocolId];
+
+  const extractedTokens = useMemo(() => {
+    return Object.keys(protocolPosition?.positionTypes || {}).reduce(
+      (acc, positionType) => {
+        acc[positionType as PositionTypeKeys] =
+          protocolPosition?.positionTypes[positionType as PositionTypeKeys]
+            ?.positions || [];
+        return acc;
+      },
+      {} as Record<PositionTypeKeys, ProtocolTokenWithMarketValue[][]>,
+    );
+  }, [protocolPosition]);
+
+  const underlyingDefiTokens = {
+    supply: useExtractUnderlyingTokens(extractedTokens.supply),
+    borrow: useExtractUnderlyingTokens(extractedTokens.borrow),
+    stake: useExtractUnderlyingTokens(extractedTokens.stake),
+    reward: useExtractUnderlyingTokens(extractedTokens.reward),
+  };
+
+  if (!protocolPosition) {
+    return <Redirect to={{ pathname: DEFAULT_ROUTE }} />;
+  }
+
+  return (
+    <Box className="main-container asset__container">
+      <Box paddingLeft={2}>
+        <Box display={Display.Flex} paddingBottom={4} paddingTop={4}>
+          <ButtonIcon
+            data-testid="defi-details-page-back-button"
+            color={IconColor.iconAlternative}
+            marginRight={1}
+            size={ButtonIconSize.Sm}
+            ariaLabel={t('back')}
+            iconName={IconName.ArrowLeft}
+            onClick={() => history.push(DEFAULT_ROUTE)}
+          />
+        </Box>
+      </Box>
+
+      <Text
+        variant={TextVariant.headingLg}
+        paddingLeft={4}
+        paddingBottom={2}
+        data-testid="defi-details-page-title"
+      >
+        {protocolPosition.protocolDetails.name}
+      </Text>
+      <Box paddingLeft={4} paddingBottom={4}>
+        <SensitiveText
+          data-testid="defi-details-page-market-value"
+          className="mm-box--color-text-alternative-soft"
+          ellipsis
+          variant={TextVariant.inherit}
+          isHidden={privacyMode}
+          length={SensitiveTextLength.Medium}
+        >
+          {'$'}
+          {protocolPosition.aggregatedMarketValue.toFixed(2)}
```

Comment:
We could probably use `formatWithThreshold` here to format consistently. If we are limited to usd support for now, we can pass the currency as a hard coded value